### ISSUE TITLE
[Request] Command to pause, but not make dialogue disappear

### ISSUE THREAD

**Issue by m-from-space** [renpy/renpy#4581]
When writing my game, I every now and then come across a situation where I want a certain event (or code) happen right after a dialogue finishes. But not in a way, that I need to set up a callback function for it.

The way I usually do it:

```
label start:
    e "Hey, I am Eileen!{nw}"
    $ renpy.notify("Something happens...")
    extend ""
```

I have two problems with that:
1. It's really ugly.
2. My translation file gets filled up with those empty `extend` commands

Now I actually cannot make the game `pause` after the notification, since it will hide the dialogue from before.

Is there already an easy solution to this problem? Am I missing something? The forum does not have an answer for this.

Suggestion:

Create a command called `wait` that just works like pause, but doesn't do anything else. (Why does pause hide the dialogue to begin with though?)

**Comment by Andykl** [comment]
I think instead of new statement maybe it is better if we make `extend$` behave like `extend ""` but does not create translation.
However I think there should be a way to do what you want with current tooling, I just don't know how.

**Comment by mal** [comment]
Are there scenarios where it does make sense for `say ""` to end up in translation files? Perhaps empty strings shouldn't be added by default, with a compat config to generate them for corner cases? That would at least remedy point 2.

If the reason you can't simply do the notify first that it has to happen on the completion of slow text? In cases with high or instant text speed there wouldn't be much difference, right? i.e.
```
label start:
    $ renpy.notify("Something happens...")
    e "Hey, I am Eileen!"
```

**Comment by mal** [comment]
With a little set up, a cleaner (at the point of use) approach is possible using the current codebase (tested on `8.0.3`). This would also avoid creating a second rollback entry for the affected line.

```rpy
init python:
    class on_slow_done:
        def __init__(self, message):
            self.message = message

        def __call__(self, event, interact=True, **kwargs):
            if event == 'slow_done':
                renpy.notify(self.message)


label start:
    'Hello'
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit.' (callback=on_slow_done('Slow text is done!'))
    'Goodbye'
    return
```

**Comment by Booplicate** [comment]
Can't you do
```renpy
e "Hey, I am Eileen!{nw}"
window show
$ renpy.notify("Something happens...")
pause
```

**Comment by Andykl** [comment]
Both `callback=` and `window show` approaches looks too wordy for such a simple thing IMO. Probably this is better.
```
def extend_notify(message):
    store._window = True
    renpy.notify(message)
    return renpy.pause()

e "Hey, I am Eileen!{nw}"
$ extend_notify("Something happens...")
```

**Comment by mal** [comment]
You'd want to swap `return renpy.pause()` for `return extend("")` to avoid the say being shown then immediately erased, at which point you wouldn't need to manipulate the window, so you could trim it down to:
```py
def extend_notify(message):
    renpy.notify(message)
    return extend("")
```
Which is essentially the original approach, but abstracted away so that it doesn't create an additional rollback entry or pollute translations.

**Comment by Booplicate** [comment]
`window show` seems to be designed for this kind of things, while `extend` seems to be more of a hack.

**Comment by mal** [comment]
`window show` is unrelated to, and solves a different problem than, that of persisting a dialogue line beyond the end of an interaction which is what's being sought here.

**Comment by Booplicate** [comment]
That seems to contradict to what you have in the documentation.
<details>
  <summary>Click</summary>
  
![1](https://user-images.githubusercontent.com/53382877/234696693-c6f87258-c1e9-4f40-baaf-244762e90151.PNG)
![2](https://user-images.githubusercontent.com/53382877/234696701-0c285b14-6734-4463-8742-f187c51f1a22.PNG)
  
</details>

**Comment by Gouvernathor** [comment]
The `window` statement is the answer to "Why does pause hide the dialogue to begin with though?".

I think extend and window are just as valid, I personally don't like to meddle with window so I'd go with this : use `$ extend("")` instead of `extend ""`, and it will do exactly the same thing without triggering the translation service.

**Comment by renpytom** [comment]
I've made it so extend "" doesn't get translated. 

That being said, I do like @mal 's callback-based approach, perhaps wrapped into a function. 

I don't think this requires further changes to Ren'Py.

**Comment by Gouvernathor** [comment]
Because if it did, you'd need a command to make dialogue disappear, maybe not after each dialogue line since every sentence replaces the former, but after each dialogue section. That would make the script longer and more verbose and so far there hasn't been much use or demand for it.

**Comment by m-from-space** [comment]
Yeah, but I mean it's okay when pause hides the dialogue window (which contains the dialogue text), but why is it making dialogue text inside the dialogue window disappear. So "window show" has no meaning in this case. Who needs empty dialogue windows? o.O

**Comment by Booplicate** [comment]
Perhaps the documentation needs a correction because to me it reads as the window management system should handle this case.

**Comment by Gouvernathor** [comment]
@Booplicate you're confusing the window and the dialogue text shown in the window. The documentation you screened is accurate and only describes how the window works.

@m-from-space I soewhat agree, personally I always disable the window as soon as the dialogue ends in my games. But this is long-standing behavior, so not easy to change.

**Comment by Booplicate** [comment]
I guess you're right. But then I fail to see the point of it - it's dialogue window, it's only useful with dialogue... So what's its point of it without the content inside?
